planner/core/task.go

```diff
@@ -1231,14 +1231,25 @@ func BuildFinalModeAggregation(
 			finalAggFunc.Mode = aggregation.CompleteMode
 		} else {
 			if aggregation.NeedCount(finalAggFunc.Name) {
-				ft := types.NewFieldType(mysql.TypeLonglong)
-				ft.Flen, ft.Charset, ft.Collate = 21, charset.CharsetBin, charset.CollationBin
-				partial.Schema.Append(&expression.Column{
-					UniqueID: sctx.GetSessionVars().AllocPlanColumnID(),
-					RetType:  ft,
-				})
-				args = append(args, partial.Schema.Columns[partialCursor])
-				partialCursor++
+				if sctx.GetSessionVars().AllowMPPExecution && finalAggFunc.Name == ast.AggFuncCount && partialIsCop {
```

Comment:
I have three questions:
1. What is this check for? please explain it in the comments.
2. As I have mentioned, **AllowMPPExecution** being true doesn't mean this function is using mpp mode. For example, user enable this var but forget adding any tiflash replica.
3. could we do count->sum converting here, instead of messing with `***_to_pb`. That's not graceful. If we have one exception check, we'd better do the weird things together.

Reply:
2. add a `mppTask` parameter to denote the specific case?
3. at early phase, I just change count->sum here, but changing the explain results.

Reply:
updated as discussed.